### ISSUE TITLE
Yubico Authenticator cannot be set as default app for nfc key

### ISSUE THREAD

**Issue by johnnydvc** [Yubico/yubioath-flutter#1311]
Since the latest update I have a couple of problems with the authenticator. The first issue ist that I was prompted to input some unlock pin that I never set before and that was not required in the older version.
I had to fix that by resetting the app and setup all my accounts again.
Another issue still remains.
On the previous app version I was used to tap my Yubikey nfc tag to the phone and the authenticato app launched directly with to my accounts. Now I'm prompted to choose between opening Firefox or the authenticator app. The problem is that I only can set Firefox as default.
That's a rather big inconvenience, since I have to tap the nfc key twice now. I used to tap and instantly see all my 2fa accounts. Now I tap to open the app and I have to tap again to open the accounts.
I cannot change this behavior in the settings of the app.
It's this intentional or maybe a problem with Firefox or android?
Thanks for any help.
![Screenshot_20231218-222222_Android_System_1](https://github.com/Yubico/yubioath-flutter/assets/4787052/f2d109e7-c6cd-4b95-86f2-7f01e910cdc9)

**Comment by AdamVe** [comment]
Hi @johnnydvc, thank you for taking the time to reach out to us. 

>  The first issue ist that I was prompted to input some unlock pin that I never set before

That should not happen - the application only asks for password if the YubiKey has a password set for the OATH application. The entered password can be "remembered" by the app so that it will not be asked again on the phone you "remember" it (but will be asked on any other device). What is also important: every YubiKey has it's own OATH password.

> Now I'm prompted to choose between opening Firefox or the authenticator app.

The NFC tap is controlled by the OS, and unfortunately, different phones and Android versions might implement this in various ways - so if you see a functionality change it could come with a system update. 
I reproduced exactly the same issue you have and will investigate if this can be fixed in any way, thanks for reporting - as a workaround, on Android 14, there is a menu Settings -> Apps -> Special app access -> Launch via NFC where app can be enabled disabled for the NFC tap. This is at least on a Pixel phone - other manufacturers might have it called in a different way.

Could you let me know what Android version do you use (eventually which manufacturer) and what versions of the app you used before the last update (was it 6.3.0 or 5.x ?)

**Comment by AdamVe** [comment]
I found another way how to make Yubico Authenticator default app for NFC Tap in Android 14 without the need to disabling NFC for other apps:
In Settings, choose _Apps_ and then _See all apps_, in the menu select _Reset app preferences_ and confirm. After tapping YubiKey, choose _Yubico Authenticator_ and _Always_ to make Yubico Authenticator open automatically.
![Screenshot_20231220-145639](https://github.com/Yubico/yubioath-flutter/assets/1954891/6f337b1b-2fb4-45ba-97c2-e3991da096e1) ![Screenshot_20231220-145342](https://github.com/Yubico/yubioath-flutter/assets/1954891/0e1cfd30-a08a-4ccc-9c4e-2677c07406fe) ![Screenshot_20231220-145354](https://github.com/Yubico/yubioath-flutter/assets/1954891/e577c0c5-7edb-4c21-befa-c992f4761e17)
![Screenshot_20231220-145406](https://github.com/Yubico/yubioath-flutter/assets/1954891/6272a787-8e93-44cf-bf1f-25cc2d763bbb) ![Screenshot_20231220-145427](https://github.com/Yubico/yubioath-flutter/assets/1954891/c2fa5a73-ef58-4313-8cac-8068ba98898a)

**Comment by johnnydvc** [comment]
Hi. Thanks for the quick reply. 
I'm on Android 13 on a Asus Zenfone 9.
Asus hasn't released Android 14 for that phone yet, but it is available as beta.
As of the yubikey authentication app, I'm usually always updated. Currently it's v 6.3.1. before it was v 6.3.0.
I did no mayor os updates in between that I can remember. 
Thanks @AdamVe for the help, unfortunately Android 13 or Asus' modified version of it has no option to reset the app defaults. Maybe it in another menu, I'll check again.

**Comment by AdamVe** [comment]
Thanks for the clarification, you are correct that the described settings for _Launch via NFC_ are Android 14 only. 

On Android 13, one can go to app information and see "Defaults" and clear those - in your case I think you should _Clear default preferences_ for Firefox and see if you can choose Yubico Authenticator for always handling the NFC taps:
![image](https://github.com/Yubico/yubioath-flutter/assets/1954891/052dafd3-11c0-4f79-90f1-25b860f6472d)

**Comment by johnnydvc** [comment]
> Thanks for the clarification, you are correct that the described settings for _Launch via NFC_ are Android 14 only.
> 
> On Android 13, one can go to app information and see "Defaults" and clear those - in your case I think you should _Clear default preferences_ for Firefox and see if you can choose Yubico Authenticator for always handling the NFC taps: ![image](https://private-user-images.githubusercontent.com/1954891/292171459-052dafd3-11c0-4f79-90f1-25b860f6472d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxODA0MTksIm5iZiI6MTcwMzE4MDExOSwicGF0aCI6Ii8xOTU0ODkxLzI5MjE3MTQ1OS0wNTJkYWZkMy0xMWMwLTRmNzktOTBmMS0yNWI4NjBmNjQ3MmQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMjFUMTczNTE5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzYxYWFhYzhiZjY1YTEwYWI2ZWQwNmRlNDE2MDIyMmE2NDM1MWE2Nzc4MjdkYmY0OGNhN2NmNzJhMDQyNGRiOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.XiIuPe6sqfbQgrQC2j_rS0a6IjDTtFUUGkl4W-w0_8A)

I've tried that, it didn't fix it. Still the same behaviour.

**Comment by AdamVe** [comment]
> I've tried that, it didn't fix it. Still the same behaviour.

I am sorry to hear that. Although I have similar experience, that one app will be stuck with "Just once" and "Always" buttons, and other apps can be run only once, I was able to clear defaults in Android 13 devices of different manufacturers (either by _Clear default preferences_ in the App Info screen or by _Reset app preferences_ in the All apps settings). 

What does not work as expected at all is the way how the system behaves after reinstalling an app, for example (versions 6.3.0 and 6.3.1 of Yubico Authenticator and Android 13 device):
1. Firefox is "Just once"/"Always" app, I uninstall it
2. After I tap YubiKey, I can choose Yubico Authenticator or Chrome to handle the NFC Tap
3. I choose Yubico Authenticator and directly select "Always"
4. Tapping NFC YubiKey opens the Yubico Authenticator
5. I install Firefox
6. I tap YubiKey

Expected:
I expect a selection where I can choose which of the three apps to use - Chrome/Firefox/Yubico Authenticator.

What really happens:
A selection opens with Firefox pre selected and having "Just once" "Always" buttons and Yubico Authenticator and Chrome listed as "Use a different app" without the possibility to make Yubico Authenticator the default. (Also important to mention, that after uninstalling Firefox, tapping NFC YubiKey opens Yubico Authenticator directly without questions).

I don't understand such behaviour and I don't recall it happening this way recently. I suspect something has changed in Android (one can get system updates through monthly Android Security updates or through Google Play Services updates). 

With that, I think I cannot do too much in the app - I will do some more investigations, maybe I get hands on an Asus Zenfone 9 to do more testing; at the end I will file a bug report at Android. Thanks for reporting this to us, @johnnydvc

**Comment by johnnydvc** [comment]
Something has happened and I don't know what. I fiddled with the settings somewhat, but I cannot say what changed it in the end. Suddenly I was able to set yubico Authenticator app as default nfc action.
Thanks for all the help
